token/cli/src/main.rs

```diff
@@ -2875,6 +2880,193 @@ async fn command_update_confidential_transfer_settings(
+        let needed_account_len =
+            ExtensionType::try_calculate_account_len::<Account>(&existing_extensions)?;
+        if needed_account_len > current_account_len {
+            token
+                .reallocate(
+                    &token_account_address,
+                    &owner,
+                    &[ExtensionType::ConfidentialTransferAccount],
+                    &bulk_signers,
+                )
+                .await?;
+        }
+    }
+
+    let res = token
+        .confidential_transfer_configure_token_account(
+            &token_account_address,
+            &owner,
+            None,
+            maximum_credit_counter,
+            elgamal_keypair,
+            aes_key,
+            &bulk_signers,
+        )
+        .await?;
+
+    let tx_return = finish_tx(config, &res, false).await?;
+    Ok(match tx_return {
+        TransactionReturnData::CliSignature(signature) => {
+            config.output_format.formatted_string(&signature)
+        }
+        TransactionReturnData::CliSignOnlyData(sign_only_data) => {
+            config.output_format.formatted_string(&sign_only_data)
+        }
+    })
+}
+
+async fn command_enable_disable_confidential_transfers(
+    config: &Config<'_>,
+    token_account_address: Pubkey,
+    owner: Pubkey,
+    bulk_signers: BulkSigners,
+    allow_confidential_credits: Option<bool>,
+    allow_non_confidential_credits: Option<bool>,
+) -> CommandResult {
+    if config.sign_only {
+        panic!("Sign-only is not yet supported.");
+    }
+
+    let account = config.get_account_checked(&token_account_address).await?;
+    let current_account_len = account.data.len();
+
+    let state_with_extension = StateWithExtensionsOwned::<Account>::unpack(account.data)?;
+    let token = token_client_from_config(config, &state_with_extension.base.mint, None)?;
+
+    let existing_extensions: Vec<ExtensionType> = state_with_extension.get_extension_types()?;
+    if !existing_extensions.contains(&ExtensionType::ConfidentialTransferAccount) {
+        panic!(
+            "Confidential transfer is not yet configured for this account. \
+        Use `configure-confidential-transfer-account` command instead."
+        );
+    }
+
+    // Reallocation (if needed)
+    let mut existing_extensions: Vec<ExtensionType> = state_with_extension.get_extension_types()?;
+    if !existing_extensions.contains(&ExtensionType::ConfidentialTransferAccount) {
+        existing_extensions.push(ExtensionType::ConfidentialTransferAccount);
+        let needed_account_len =
+            ExtensionType::try_calculate_account_len::<Account>(&existing_extensions)?;
+        if needed_account_len > current_account_len {
+            token
+                .reallocate(
+                    &token_account_address,
+                    &owner,
+                    &[ExtensionType::ConfidentialTransferAccount],
+                    &bulk_signers,
+                )
+                .await?;
+        }
+    }
```

Comment:
Considering the command errors if it doesn't already contain the extension right before this, it's probably best to remove this logic since it'll never get hit

Reply:
Oops, thanks for the catch! Removed.